### ISSUE TITLE
Could not train well on small dataset

### ISSUE THREAD

**Issue by linghai06** [bearpaw/pytorch-pose#36]
Hi bearpaw, I tried the codes with default setting on small dataset, say 4 images and found that it could not train better than using full mpii dataset, with respect to training loss, i.e. the training loss on full mpii dataset is even lower(~ 1e-4) while on a small set of 4 images it's been on a plateau at  ~1e-3.

It is very strange. As I think verifying the model and the codes could overfit a small set is necessary before any further step, could you provide some advise for above issue? Thanks.

**Comment by linghai06** [comment]
To be more concrete, modified codes and training scripts are attached as follows:
![image](https://user-images.githubusercontent.com/7567001/34859042-d5527a66-f78e-11e7-9f85-402ca96a243a.png)

> self.train, self.valid = self.train[:4], self.valid[:4]
is added in datasets/mpii.py to select only 4 images

And here is the scripts

![image](https://user-images.githubusercontent.com/7567001/34859077-0931c800-f78f-11e7-9bc6-114ab072ba5b.png)

**Comment by bearpaw** [comment]
I think 4 images are not sufficient to train a pose estimator.

**Comment by linghai06** [comment]
Well, literally it is not training in normal sense but testing if the gradient flow could drive the network to memorize such a simple setting. I think it is a verification procedure before any further step.

There is a similar discussion on reddit: https://www.reddit.com/r/MachineLearning/comments/5pidk2/d_is_overfitting_on_a_very_small_data_set_a/****

**Comment by bearpaw** [comment]
Interesting. What is the batchsize you used during training and testing? It is possible that the BN is not trained properly.

**Comment by linghai06** [comment]
Only 4 images were used and the batch-size was set to 4. Currently I only pay attention to the training loss for this 4 images and surprisingly found that it dropped slower than using full mpii and also stopped at a plateau higher.
